### PR TITLE
Enha: CI Update sentry-cli

### PR DESCRIPTION
**Pull request by denrase**
_#skip-changelog_

- [x] ~~Blocked: Need to merge #143 first.~~
- [x] Wait until https://github.com/getsentry/github-workflows/pull/18 is merged.

### CODE REVIEW

File: .github/workflows/update-deps.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+name: Update Dependencies
+
+on:
+  schedule:
+    - cron: '0 3 * * *'
```
Comment (philipphofmann) [inline .github/workflows/update-deps.yml:5]: `l`: Please add a comment when this runs. I think it's every day at 3 am UTC?

**Review by philipphofmann — COMMENTED** [review]
Nice, what's missing to finish this PR?

File: .github/workflows/update-deps.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
+name: Update Dependencies
+
+on:
+  # Run every day.
+  schedule:
+    - cron: '0 3 * * *'
+  # And on on every PR merge so we get the updated dependencies ASAP, and to make sure the changelog doesn't conflict.
+  push:
+    branches:
+      -master
```
Comment (vaind) [inline .github/workflows/update-deps.yml:10]:
```suggestion
      - master
```

**Comment by denrase** [comment]
@philipphofmann Alright, we now also automatically update and commit the binaries when an PR for a new `sentry-cli` version is created. We are using a fork of the workflow and need to wait until it is merged to finish this PR here.

File: .github/workflows/update-deps.yml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+jobs:
+  update-cli:
+    uses: denrase/github-workflows/.github/workflows/updater.yml@9f98d2bca80719220d03f18740c052b9ae6ebbaa # TODO: Use sentry workflow when merged.
+    with:
+      path: script/sentry-cli.properties
+      name: CLI
+    secrets:
+      api_token: ${{ secrets.CI_DEPLOY_KEY }}
+  update-binaries:
+    runs-on: macos-latest
```
Comment (vaind) [inline .github/workflows/update-deps.yml:21]: btw ubuntu runners are more readily available and faster to start up, most of the times.

File: .github/workflows/update-deps.yml [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  schedule:
+    - cron: '0 3 * * *'
+  # And on on every PR merge so we get the updated dependencies ASAP, and to make sure the changelog doesn't conflict.
+  push:
+    branches:
+      -master
+
+jobs:
+  update-cli:
+    uses: denrase/github-workflows/.github/workflows/updater.yml@9f98d2bca80719220d03f18740c052b9ae6ebbaa # TODO: Use sentry workflow when merged.
```
Comment (vaind) [inline .github/workflows/update-deps.yml:14]: v1 is updated
```suggestion
    uses: denrase/github-workflows/.github/workflows/updater.yml@v1
```

**Review by vaind — CHANGES_REQUESTED** [review]

File: .github/workflows/update-deps.yml [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+      path: script/sentry-cli.properties
+      name: CLI
+    secrets:
+      api_token: ${{ secrets.CI_DEPLOY_KEY }}
+  update-binaries:
+    runs-on: ubuntu-latest
+    needs: update-cli
+    steps:
+      - name: Checkout PR branch
+        if: needs.update-cli.outputs.originalTag != needs.update-cli.outputs.latestTag
```
Comment (vaind) [inline .github/workflows/update-deps.yml:25]: just wondering - can this condition be moved to the job? I've never used it that way but seems to be possible: https://docs.github.com/en/actions/using-jobs/using-conditions-to-control-job-execution

Comment (denrase) [reply .github/workflows/update-deps.yml:25]: Yeah sounds good, less duplication. 👍

**Review by brustolin — APPROVED** [review]
LGTM

**Pull request merged**